src/core/lib/transport/metadata_batch.h

```diff
@@ -184,6 +187,49 @@ struct GrpcTagsBinMetadata : public SimpleSliceBasedMetadata {
   static absl::string_view key() { return "grpc-tags-bin"; }
 };
 
+template <typename Int>
+struct SimpleIntBasedMetadataBase {
+  using ValueType = Int;
+  using MementoType = Int;
+  static ValueType MementoToValue(MementoType value) { return value; }
+  static Slice Encode(ValueType x) {
+    char tmp[GPR_LTOA_MIN_BUFSIZE];
+    gpr_ltoa(x, tmp);
```

Comment:
Suggest using `absl::StrCat(x)`, since we can pass the result of that directly to `Slice::FromCopiedString()`.

Reply:
See above.